--- cfitsio-3.48/CMakeLists.txt
+++ cfitsio-3.48/CMakeLists.txt
@@ -58,9 +58,7 @@
 FILE(GLOB H_FILES "*.h")
 
 IF (USE_PTHREADS)
-    FIND_PACKAGE(pthreads REQUIRED)
-    INCLUDE_DIRECTORIES(${PTHREADS_INCLUDE_DIR})
-    ADD_DEFINITIONS(-D_REENTRANT)
+    FIND_PACKAGE(Threads REQUIRED)
 ENDIF()
 
 # Math library (not available in MSVC or MINGW)
@@ -71,7 +69,7 @@
 ENDIF()
 
 # Support for remote file drivers is not implemented for native Windows:
-IF (NOT MSVC)
+IF (NOT WIN32)
     # Find library needed for gethostbyname:
     CHECK_FUNCTION_EXISTS("gethostbyname" CMAKE_HAVE_GETHOSTBYNAME)
     IF(NOT CMAKE_HAVE_GETHOSTBYNAME)
@@ -132,7 +130,10 @@
 ENDIF()
 
 ADD_LIBRARY(${LIB_NAME} ${LIB_TYPE} ${H_FILES} ${SRC_FILES})
-TARGET_LINK_LIBRARIES(${LIB_NAME} ${PTHREADS_LIBRARY} ${M_LIB} ${CURL_LIBRARIES})
+TARGET_LINK_LIBRARIES(${LIB_NAME} Threads::Threads ${M_LIB} ${CURL_LIBRARIES} z)
+if(WIN32)
+    target_compile_options(${LIB_NAME} PUBLIC -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -D_REENTRANT)
+endif()
 
 SET_TARGET_PROPERTIES(${LIB_NAME} PROPERTIES VERSION ${${PROJECT_NAME}_VERSION} SOVERSION ${${PROJECT_NAME}_MAJOR_VERSION})
 install(TARGETS ${LIB_NAME} DESTINATION ${LIB_DESTINATION})

--- cfitsio-3.48/cfitsio.pc.cmake
+++ cfitsio-3.48/cfitsio.pc.cmake
@@ -6,7 +6,7 @@
 Name: cfitsio
 Description: FITS File Subroutine Library
 URL: https://heasarc.gsfc.nasa.gov/fitsio/
-Version: @CFITSIO_MAJOR@.@CFITSIO_MINOR@
-Libs: -L${libdir} @PKG_CONFIG_LIBS@
-Libs.private: -lm
+Version: @PROJECT_NAME_VERSION@
+Libs: -L${libdir} -l@LIB_NAME@
+Libs.private: -lm -lz
 Cflags: -I${includedir}
 